eng/pipelines/pool-migration.yml

```diff
@@ -0,0 +1,98 @@
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-ios
+      ExcludePaths: "^./sdk/|^./eng/common/"
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-java
+      ExcludePaths: "^./sdk/|^./eng/common/|/resources/session-records/"
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-js
+      ExcludePaths: "^./sdk/|^./eng/common/|/recordings/node/"
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-net
+      ExcludePaths: "^./sdk/|^./eng/common/|/tests/SessionRecords/"
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+    - RepoOwner: Azure
+      RepoName: azure-sdk-for-python
+      ExcludePaths: "^./sdk/|^./eng/common/|/tests/recordings/"
+      IncludePaths: "^./sdk/.*/platform-matrix.*.json$|^./sdk/.*/.*yml$"
+# The input parameter format is like:
+# - MigrateFrom: azsdk-pool-mms-win-2019-general
+#   MigrateTo: azsdk-pool-mms-win-2022-general
+# - MigrateFrom: windows-2019
+#   MigrateTo: windows-2022
+# - MigrateFrom: windows2019
+#   MigrateTo: windows2022
+- name: MigrationMap
+  type: object
+  default: []
+trigger: none
+pr: none
+
+jobs:
+  - job: Migration
+    pool:
+      name: azsdk-pool-mms-ubuntu-2004-general
+    variables:
+      - template: ./templates/variables/globals.yml
+    steps:
+      - ${{ each repo in parameters.Repos }}:
+        - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+          parameters:
+            SkipCheckoutNone: true
+            Repositories:
+              - Name: ${{ repo.RepoOwner }}/${{ repo.RepoName }}
+                WorkingDirectory: $(Build.SourcesDirectory)/docs/${{ repo.RepoName }}
+            Paths:
+              - /*
+        - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+        - pwsh: |
+            $migrationMap = '${{ convertToJson(parameters.MigrationMap) }}'
+            $(Build.SourcesDirectory)/eng/scripts/Repo-File-Content-Replacements.ps1 `
+              -ExcludePathsRegex '${{ repo.ExcludePaths }}' `
+              -IncludePathsRegex '${{ repo.IncludePaths }}' `
+              -MigrationMap $migrationMap
+          displayName: Update pool version for "${{ repo.RepoName }}"
+          workingDirectory: $(Build.SourcesDirectory)/docs/${{ repo.RepoName }}
+          continueOnError: true
+        - template: /eng/common/pipelines/templates/steps/create-pull-request.yml
+          parameters:
+            BaseBranchName: $(DefaultBranch)
+            PRBranchName: migrate-pool
+            CommitMsg: "Migrate to new pool"
+            RepoOwner: ${{ repo.RepoOwner }}
+            RepoName: ${{ repo.RepoName }}
+            PRTitle: "Migrate to new pool"
```

Comment:
Can we pass this in as a UI parameter? It would be nice to enable more descriptive commit messages so we can find it later. I was trying to search open/closed PRs for text like "windows" and "2022" but couldn't find anything because of the non-specific title.

Also, perhaps use the commit message and pr title from the same value?

Reply:
It would also be good to pass a PR description linking back to the build that generated it. Otherwise it can be confusing figuring out where the PRs came from and for what reason and who invoked it.

Reply:
We need to add extra short description (parameters.From, parameters.To) here as the migration map including more than 1 migration pair.

Reply:
> It would also be good to pass a PR description linking back to the build that generated it. Otherwise it can be confusing figuring out where the PRs came from and for what reason and who invoked it.

Usually, we triggered the pipeline directly without a PR, so linked back to PR seems an redundant step here.

Reply:
I mean to say a link in the generated PRs to the build that created them, not the other way around.

Reply:
Is there any concern of adding internal pipeline to public PR? If not, I think it is doable.

Reply:
Adding 2 parameters: `ShortForMigrateFrom`, `ShortForMigrateTo`

Reply:
<img width="809" alt="image" src="https://user-images.githubusercontent.com/48036328/210444441-83767a35-4374-47da-9b8e-4498e3e8ad14.png">
Here is the description in the PR body which linked back to the pipeline.

Updated the description to "This is auto PR generated by pipeline: ${link}"